src/adapters/node-http/core.ts

```diff
@@ -42,12 +41,8 @@ export const createOpenApiNodeHttpHandler = <
 ) => {
   const router = cloneDeep(opts.router);
 
-  // Validate router
-  generateOpenApiDocument(router, { title: '-', version: '-', baseUrl: '-' });
```

Comment:
With `nextjs` this will run on every request - can be very slow depending on the number of procedures, hence removed. This shouldn't be an issue though because you will likely run `generateOpenApiDocument` elsewhere in your app which will throw any errors.

Reply:
This could be a best-of-both-worlds solution

```ts
if (process.env.NODE_ENV !== 'production') {
  generateOpenApiDocument(router, { title: '-', version: '-', baseUrl: '-' });
}
```